homeassistant/components/qnap/__init__.py

```diff
@@ -1 +1,36 @@
 """The qnap component."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+
+from .const import DOMAIN
+from .coordinator import QnapCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set the config entry up."""
+    hass.data.setdefault(DOMAIN, {})
+    coordinator = QnapCoordinator(hass, config_entry)
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_config_entry_first_refresh()
+    if not coordinator.last_update_success:
+        raise PlatformNotReady
```

Comment:
These two lines are not needed anymore. `async_config_entry_first_refresh` is sufficient on it's own.
```suggestion
```